boot/assets.go

```
@@ -570,8 +572,14 @@ func (o *TrustedAssetsUpdateObserver) observeRollback(bl bootloader.Bootloader,
 
 // BeforeWrite is called when the update process has been staged for execution.
 func (o *TrustedAssetsUpdateObserver) BeforeWrite() error {
-	// TODO:UC20:
-	// - reseal with a given state of modeenv
+	if o.modeenv == nil {
+		// modeenv wasn't even loaded yet, meaning none of the trusted
+		// boot assets was updated
+		return nil
+	}
+	if err := resealKeyToModeenv(o.model, o.modeenv); err != nil {
+		return fmt.Errorf("cannot reseal encryption key: %v", err)
```

this should just return the error, or will get repetitive messages